### PR TITLE
Bug 1407315, 1406926, 1407231, 1406924 - Fix a bunch of visual regressions 

### PR DESCRIPTION
**Pull request by farhanpatel**
@garvankeeley You were right about canceled animations. I had accounted for making sure the icon looked right after a canceled animation but forgot about what would happen after you tapped it. 🤦‍♂️

### CODE REVIEW

**Comment by garvankeeley** [comment]
It was just based on a guess that something might not work, 100% of the time I have tried to do something based on that cancel flag in the animation completion block it hasn’t worked quite right me. I will count this as 101% now ;).

**Comment by farhanpatel** [comment]
I opted to make this into one PR in order to not have to request 4 different reviewers.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 1]
```diff
@@ -612,6 +612,7 @@ class BrowserViewController: UIViewController {
         [header, footer, readerModeBar].forEach { view in
                 view?.transform = CGAffineTransform.identity
         }
+        self.statusBarOverlay.isHidden = false
```
Comment (garvankeeley) [inline Client/Frontend/Browser/BrowserViewController.swift]: nit: `self`. I actually have to expand the github view when I see this to confirm the code is not in a closure.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 2]
```diff
@@ -2845,6 +2846,7 @@ extension BrowserViewController: Themeable {
         let ui: [Themeable?] = [urlBar, toolbar, readerModeBar, topTabsViewController]
         ui.forEach { $0?.applyTheme(themeName) }
         statusBarOverlay.backgroundColor = shouldShowTopTabsForTraitCollection(traitCollection) ? UIColor(rgb: 0x272727) : urlBar.backgroundColor
+        self.setNeedsStatusBarAppearanceUpdate()
```
Comment (garvankeeley) [inline Client/Frontend/Browser/BrowserViewController.swift]: nit: self

File: Client/Frontend/Browser/TabTrayController.swift [hunk 3]
```diff
@@ -968,10 +968,8 @@ extension TabTrayController: UIViewControllerPreviewingDelegate {
     func previewingContext(_ previewingContext: UIViewControllerPreviewing, commit viewControllerToCommit: UIViewController) {
         guard let tpvc = viewControllerToCommit as? TabPeekViewController else { return }
         tabManager.selectTab(tpvc.tab)
-        _ = self.navigationController?.popViewController(animated: true)
-
+        self.navigationController?.popViewController(animated: true)
```
Comment (garvankeeley) [inline Client/Frontend/Browser/TabTrayController.swift]: nit: self

File: Client/Frontend/Widgets/TabsButton.swift [hunk 4]
```diff
@@ -215,7 +215,8 @@ class TabsButton: UIButton {
             }
             
             let completion: (Bool) -> Void = { completed in
-                if completed {
+                let noActiveAnimations = self.insideButton.layer.animationKeys()?.isEmpty ?? true
```
Comment (garvankeeley) [inline Client/Frontend/Widgets/TabsButton.swift]: Interesting :)! Did not know of this method.

**Pull request merged**